### PR TITLE
ci: add slack notifications

### PR DESCRIPTION
**Pull request by jrusz**
This pull request includes:

Trying out how slack notifications about nightly pipeline execution could work.

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by jrusz** [comment]
I did some testing and it works as intended which is to either report failure or success of the scheduled nightly pipeline to our slack channel. 
Example run of failure: https://gitlab.com/osbuild/ci/osbuild-composer/-/pipelines/334477690
Example run of success: https://gitlab.com/osbuild/ci/osbuild-composer/-/pipelines/334479852
I tested it in #bot-testing channel, this is how it's going to look: 
image
image

**Comment by atodorov** [comment]
In the previous Telegram implementation we had the COMPOSE_ID integrated in the message. Can you either add that or the entire COMPOSE_URL to easily distinguish what the reported status is for.

File: schutzbot/slack_notification.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+set -eux
+
+MESSAGE="\"Nightly pipeline execution finished with status *$1* $2 \n Link to results: $CI_PIPELINE_URL \""
+
+curl \
+    -X POST \
+    -H 'Content-type: application/json' \
+    --data '{"text": "test", "blocks": [ { "type": "section", "text": {"type": "mrkdwn", "text":'"$MESSAGE"'}}]}' \
+    "$WEBHOOK_URL"
```
Comment (atodorov) [inline schutzbot/slack_notification.sh:11]: This variable can use a better name, like SLACK_WEBHOOK_URL

**Review by atodorov — CHANGES_REQUESTED** [review]
Minor changes requested, otherwise looks good.

**Comment by atodorov** [comment]
> #bot-testing channel

How/where do you control in which channel messages are sent ?

**Comment by jrusz** [comment]
> In the previous Telegram implementation we had the COMPOSE_ID integrated in the message. Can you either add that or the entire COMPOSE_URL to easily distinguish what the reported status is for.

I'll try, I'm not sure from where to get it in that stage since I don't think it's in any of CI env vars, but maybe I could parse it from some artifacts. 

>How/where do you control in which channel messages are sent ? Ideally we'd like for this to be in source code so that anyone can change it if necessary.

The `WEBHOOK_URL` will be defined as a scheduled pipeline variable and this url is taken from the slack app settings. It seems  that I can add collaborators there, I'll try it and send you a link over chat.

**Review by atodorov — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -236,6 +237,25 @@ API:
         INTERNAL_NETWORK: ["true"]
         DISTRO_CODE: ["rhel_90"]
 
+NIGHTLY_FAIL:
+  stage: slack-report
```
Comment (ondrejbudai) [inline .gitlab-ci.yml:241]: Can we please move the new stages into the finish state? That stage is used for status reporting.

Comment (jrusz) [reply .gitlab-ci.yml]: Yeah sure, didn't realize I could use that instead of creating another stage :)

**Review by ondrejbudai — APPROVED** [review]
Looks good, merged whenever you want. :)

**Comment by atodorov** [comment]
> Looks good, merged whenever you want. :)

@ondrejbudai I don't have permissions to click the Merge button anymore, probably @jrusz doesn't have either. Something has changed recently so please merge this PR.

**Comment by ondrejbudai** [comment]
Can you try that again?

**Pull request merged**